### ISSUE TITLE
Upgrade claims installer type changed while it hasn't

### ISSUE THREAD

**Issue by BraveThrasher** [microsoft/winget-cli#3295]
### Brief description of your issue

When trying to upgrade the package "BelgianGovernment.eIDViewer" from version 5.1.5998 to version 5.1.6019, winget claims that the installer type changed.
However when querying the information of both versions it clearly shows the same installer type. See screenshot below
![image](https://github.com/microsoft/winget-cli/assets/3652075/499a2571-2b12-4969-aa54-5f7a3acea830)

### Steps to reproduce

1. Install version 5.1.5998 of "BelgianGovernment.eIDViewer"
2. Try upgrading to version 5.1.6019 (currently the latest version)

### Expected behavior

The upgrade to be installed successfully.

### Actual behavior

The upgrade fails with the text:

> A newer version was found, but the install technology is different from the current version installed. Please uninstall the package and install the newer version.

### Environment

```shell
Windows Package Manager v1.4.10173
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.19045.2965
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.19.10173.0
```

**Comment by Trenly** [comment]
Looks like the installer actually extracts and runs an MSI, which causes the mismatch. I'll try updating the manifest

**Comment by Trenly** [comment]
@denelon - The AppsAndFeaturesEntries have been updated so that it should expect the MSI entry type, but the issue is still occurring. Probably an edge case that wasn't considered during the initial implementation.

**Comment by denelon** [comment]
Does it have a productCode or upgradeCode?

**Comment by Trenly** [comment]
> Does it have a productCode or upgradeCode?

I don't believe the AppsAndFeaturesEntries have the product code specified, but I did specify `InstallerType: msi` in the AppsAndFeaturesEntries

**Comment by denelon** [comment]
OK, I'll see if I can find someone else to look into what's going on here.
